e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,187 @@
+		"cpe:/a:redhat:enterprise_linux:8::baseos", "cpe:/a:redhat:rhel:8.3::baseos",
+	}
+	return &v1.GetNodeVulnerabilitiesRequest{
+		OsImage:          "Red Hat Enterprise Linux CoreOS 45.82.202008101249-0 (Ootpa)",
+		KernelVersion:    "0.0.1",
+		KubeletVersion:   "0.0.1",
+		KubeproxyVersion: "0.0.1",
+		Runtime: &v1.GetNodeVulnerabilitiesRequest_ContainerRuntime{
+			Name:    "docker",
+			Version: "0.0.1",
+		},
+		Notes: notes,
+		Components: &v1.Components{
+			Namespace: "rhcos:4.11",
+			RhelComponents: []*v1.RHELComponent{
+				{
+					Id:        int64(1),
+					Name:      "libksba",
+					Namespace: "rhel:8",
+					Version:   "1.3.5-7.el8",
+					Arch:      "x86_64",
+					Module:    "", // must be empty, otherwise scanner does not return any vulns
+					Cpes:      cpes,
+					AddedBy:   "",
+				},
+				{
+					Id:        int64(2),
+					Name:      "tar",
+					Namespace: "rhel:8",
+					Version:   "1.27.1.el8",
+					Arch:      "x86_64",
+					Module:    "",
+					Cpes:      cpes,
+					AddedBy:   "",
+				},
+				{
+					Id:        int64(3),
+					Name:      "grep",
+					Namespace: "rhel:8",
+					Version:   "3.1-6.el8",
+					Arch:      "x86_64",
+					Module:    "",
+					Cpes:      cpes,
+					AddedBy:   "",
+				},
+			},
+			LanguageComponents: nil,
+		},
+	}
+}
+
+func TestGRPCGetRHCOSNodeVulnerabilities(t *testing.T) {
+	conn := connectToScanner(t)
+	client := v1.NewNodeScanServiceClient(conn)
+
+	cases := []struct {
+		name             string
+		request          *v1.GetNodeVulnerabilitiesRequest
+		responseContains *v1.GetNodeVulnerabilitiesResponse
+	}{
+		{
+			name:    "Selected vulnerabilities should be returned by the certified scan",
+			request: buildRequestCase([]v1.Note{}),
+			responseContains: &v1.GetNodeVulnerabilitiesResponse{
+				// We conduct a spot-checking here - more vulns can be returned from scanner for libksba and tar,
+				// but we care only about the selected one as it is sufficient for this test case.
+				// (We do not test that the set of vulns is complete, we test that the API returns any vulns if expected)
+				Features: []*v1.Feature{
+					{
+						Name:            "libksba",
+						Version:         "1.3.5-7.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnLibksba},
+					},
+					{
+						Name:            "tar",
+						Version:         "1.27.1.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{vulnTar},
+					},
+					{
+						Name:            "grep",
+						Version:         "3.1-6.el8.x86_64",
+						Vulnerabilities: []*v1.Vulnerability{},
+					},
+				},
+				NodeNotes: nil,
+			},
+		},
+		{
+			name:    "Uncertified scan is unsupported for RHCOS and returns no features",
+			request: buildRequestCase([]v1.Note{v1.Note_CERTIFIED_RHEL_SCAN_UNAVAILABLE}),
+			responseContains: &v1.GetNodeVulnerabilitiesResponse{
+				Features:  []*v1.Feature{},
+				NodeNotes: nil,
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			c := c
```

Comment:
I don't think this is needed when it's not a pointer, right?

Reply:
`func(t *testing.T)` does not accept `c` as a param, so this is needed. 
It is not the best explanation of the problem, but I usually share this link as a reference material http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/index.html#closure_for_it_vars

Reply:
The page fails to call out explicitly is that this matters when you start goroutines (or the code that's called starts goroutines). We don't do start goroutines here and can get by without `c := c`.

Reply:
This is affected not only by Goroutines but also by closures. If we ever decide to run tests in parallel, then only one test case will be run and this bug will be difficult to find out. 

https://go.dev/play/p/HDF-sPHnTjd (comment and uncomment the `c :=c` line)

I know that probability of running into this issue is low, but there is no harm in keeping this line.
Hence, I would like to keep it for now, but I am open to discussing this topic when needed.